It's now failing because it lacks the WINUI part.. can you do it ? 

```
  D:\agent\1\s\src\Core\src\Handlers\Entry\EntryHandler.cs(15,45): error CS0103: The name 'MapVerticalTextAlignment' does not exist in the current context [D:\agent\1\s\src\Core\src\Core-net6.csproj]
```

--- a/src/Core/src/Core/ITextAlignment.cs
+++ b/src/Core/src/Core/ITextAlignment.cs
@@ -9,5 +9,10 @@ namespace Microsoft.Maui
 		/// Gets the horizontal text alignment.
 		/// </summary>
 		TextAlignment HorizontalTextAlignment { get; }
+
+		/// <summary>
+		/// Gets the vertical text alignment.
+		/// </summary>
+		TextAlignment VerticalTextAlignment { get; }
 	}
 }

--- a/src/Core/src/Handlers/Entry/EntryHandler.Android.cs
+++ b/src/Core/src/Handlers/Entry/EntryHandler.Android.cs
@@ -93,6 +93,11 @@ namespace Microsoft.Maui.Handlers
 			handler.NativeView?.UpdateHorizontalTextAlignment(entry);
 		}
 
+		public static void MapVerticalTextAlignment(EntryHandler handler, IEntry entry)
+		{
+			handler?.NativeView?.UpdateVerticalTextAlignment(entry);
+		}
+
 		public static void MapIsTextPredictionEnabled(EntryHandler handler, IEntry entry)
 		{
 			handler.NativeView?.UpdateIsTextPredictionEnabled(entry);

--- a/src/Core/src/Handlers/Entry/EntryHandler.Standard.cs
+++ b/src/Core/src/Handlers/Entry/EntryHandler.Standard.cs
@@ -10,6 +10,7 @@ namespace Microsoft.Maui.Handlers
 		public static void MapTextColor(IViewHandler handler, IEntry entry) { }
 		public static void MapIsPassword(IViewHandler handler, IEntry entry) { }
 		public static void MapHorizontalTextAlignment(IViewHandler handler, IEntry entry) { }
+		public static void MapVerticalTextAlignment(IViewHandler handler, IEntry entry) { }
 		public static void MapIsTextPredictionEnabled(IViewHandler handler, IEntry entry) { }
 		public static void MapMaxLength(IViewHandler handler, IEntry entry) { }
 		public static void MapPlaceholder(IViewHandler handler, IEntry entry) { }

--- a/src/Core/src/Handlers/Entry/EntryHandler.cs
+++ b/src/Core/src/Handlers/Entry/EntryHandler.cs
@@ -12,6 +12,7 @@ namespace Microsoft.Maui.Handlers
 			[nameof(IEntry.ClearButtonVisibility)] = MapClearButtonVisibility,
 			[nameof(IEntry.Font)] = MapFont,
 			[nameof(IEntry.HorizontalTextAlignment)] = MapHorizontalTextAlignment,
+			[nameof(IEntry.VerticalTextAlignment)] = MapVerticalTextAlignment,
 			[nameof(IEntry.IsPassword)] = MapIsPassword,
 			[nameof(IEntry.IsReadOnly)] = MapIsReadOnly,
 			[nameof(IEntry.IsTextPredictionEnabled)] = MapIsTextPredictionEnabled,

--- a/src/Core/src/Handlers/Entry/EntryHandler.iOS.cs
+++ b/src/Core/src/Handlers/Entry/EntryHandler.iOS.cs
@@ -68,6 +68,11 @@ namespace Microsoft.Maui.Handlers
 			handler.NativeView?.UpdateHorizontalTextAlignment(entry);
 		}
 
+		public static void MapVerticalTextAlignment(EntryHandler handler, IEntry entry)
+		{
+			handler?.NativeView?.UpdateVerticalTextAlignment(entry);
+		}
+
 		public static void MapIsTextPredictionEnabled(EntryHandler handler, IEntry entry)
 		{
 			handler.NativeView?.UpdateIsTextPredictionEnabled(entry);

--- a/src/Core/src/Platform/Android/EditTextExtensions.cs
+++ b/src/Core/src/Platform/Android/EditTextExtensions.cs
@@ -59,6 +59,11 @@ namespace Microsoft.Maui
 			editText.UpdateHorizontalAlignment(entry.HorizontalTextAlignment, editText.Context != null && editText.Context.HasRtlSupport());
 		}
 
+		public static void UpdateVerticalTextAlignment(this AppCompatEditText editText, IEntry entry)
+		{
+			editText.UpdateVerticalAlignment(entry.VerticalTextAlignment);
+		}
+
 		public static void UpdateIsTextPredictionEnabled(this AppCompatEditText editText, IEntry entry)
 		{
 			editText.SetInputType(entry);

--- a/src/Core/src/Platform/iOS/TextFieldExtensions.cs
+++ b/src/Core/src/Platform/iOS/TextFieldExtensions.cs
@@ -39,6 +39,11 @@ namespace Microsoft.Maui
 			textField.TextAlignment = textAlignment.HorizontalTextAlignment.ToNative(true);
 		}
 
+		public static void UpdateVerticalTextAlignment(this UITextField textField, ITextAlignment textAlignment)
+		{
+			textField.TextAlignment = textAlignment.VerticalTextAlignment.ToNative(true);
+		}
+
 		public static void UpdateIsTextPredictionEnabled(this UITextField textField, IEntry entry)
 		{
 			if (entry.IsTextPredictionEnabled)

